### PR TITLE
Improve the default styling for the SearchBar component.

### PR DESCRIPTION
**Pull request by tmeyer2115**
This CR uses the updated mocks to improve the default styling of the SearchBar
component. Some default styling for AutoComplete results was also updated.
Specifically, the updates made were:

1. Update the color of the SearchBar container's outline and the Autocomplete's
   separator.
2. Update the shadow effect when focusing and hovering over the SearchBar
   container.
3. Update the border of the SearchBar container.
4. Update font size and color for the 'Search' and 'Clear Search' buttons.
   Removed unnecessary hover and focus styling for these buttons too.
5. Do not have the Autocomplete separator when on mobile.
6. Remove border around SearchBar container when on mobile.
7. Adjust padding between the end of the SearchBar and text input.

J=SPR-1491
TEST=auto, manual

Reran all Jest tests. Verified each of the style updates manually and confirmed
their adherance with the mock. Tested with both a mobile version and desktop
version of an Answers experience.

### CODE REVIEW

**Review by Mxs2019 — APPROVED** [review]
LGTM

File: src/ui/sass/modules/_SearchBar.scss [hunk 1]
```diff
@@ -81,28 +81,22 @@
...
+    font: inherit;
     padding-left: $base-spacing;
-    padding-right: $base-spacing;
     padding-top: $base-spacing / 2;
     padding-bottom: $base-spacing / 2;
     background-color: $searchbar-button-background-color-base;
-    color: $searchbar-button-text-color-base;
-
-    &:hover,
-    &:focus
```
Comment (benmcginnis) [inline src/ui/sass/modules/_SearchBar.scss:95]: so it's been determined that we don't need a focus state for the buttons?

Comment (tmeyer2115) [reply src/ui/sass/modules/_SearchBar.scss:95]: Added back the focus state styling for the buttons.

**Review by benmcginnis — APPROVED** [review]
lgtm

**Pull request merged**